### PR TITLE
[ETCM-2928] Document usage of CTL Scripts

### PR DESCRIPTION
**Pull request by AmbientTea**

### CODE REVIEW

File: CTL_Scripts_Reference.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    "ogmiosDatumCache": {
+      "host": "localhost",
+      "port": 9999,
+      "path": null,
+      "secure": false
+    }
+  },
+  "sidechainParameters": {
+    "chainId": 123, // --sidechain-id
+    "genesisHash": "0x000", // --sidechain-genesis-hash
```
Comment (LGLO) [inline CTL_Scripts_Reference.md:81]: Late spot: there should be no `0x` in front of hex string. I also think we can just copy there something real like:
`"5f4a33beaf8876b9697fb567c9af11390434086a10454c32091e2017366dcd7a"`

File: CTL_Scripts_Reference.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    "genesisMint": null, // --genesis-mint-utxo
+    "genesisUtxo": "0x000", // --genesis-committee-hash-utxo
```
Comment (LGLO) [inline CTL_Scripts_Reference.md:83]: UTXO format is different, like `"9713f220d6d388a63fe3f426212da3af063372215e8e764c470225eab0384556#1"` (there is output index)

File: CTL_Scripts_Reference.md (hunk 1 above)
Comment (AmbientTea) [reply CTL_Scripts_Reference.md]: done

File: CTL_Scripts_Reference.md (hunk 2 above)
Comment (AmbientTea) [reply CTL_Scripts_Reference.md]: Changed

**Review by LGLO — APPROVED** [review]

File: CTL_Scripts_Reference.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+### Register committee candidate
+
+In order to generate the signatures, you can use the signature generator tool:
+
+```shell
+cabal run trustless-sidechain-gen-signatures -- \
+  --spo-signing-key fa832cc7ad4a0990f36db287df51c62a64c12287e161c07fbc8a4bde0b587c0a \
+  --sidechain-signing-key fa832cc7ad4a0990f36db287df51c62a64c12287e161c07fbc8a4bde0b587c0a \
+  --registration-utxo 7eddcb7807899d5078ebc25c59d372b484add88604db461e6ef077fd0379733d#0
+```
```
Comment (AmbientTea) [inline CTL_Scripts_Reference.md:130]: This was copied from the trustless-sidechain repo's documentation, but I don't think we can use this here, becuse we're not publishing the trustless sidechain project yet and this is not available in the ctl scripts. Do we have any other way to generate the signatures?

**Comment by drospa** [comment]
@AmbientTea  We shall not use FUEL and replace it with SC_Token. This is according to legal change of names.

File: CTL_Scripts_Reference.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+node main.js -- burn \
+  --payment-signing-key-file $SIGNING_KEY \
+  --genesis-committee-hash-utxo df24e6edc13440da24f074442a858f565b5eba0a9c8d6238988485a3ed64cf1f#0 \
+  --genesis-mint-utxo df24e6edc13440da24f074442a858f565b5eba0a9c8d6238988485a3ed64cf1f#0 \
+  --sidechain-id 1 \
+  --sidechain-genesis-hash 112233 \
+  --threshold 2/3 \
+  --amount 5 \
+  --recipient aabbcc
+```
```
[the remark this answers is not in the document]
Comment (AmbientTea) [reply CTL_Scripts_Reference.md:119]: Unfortunately, I'm still having problems with nix when trying to set up an environment to get example responses.

File: CTL_Scripts_Reference.md (hunk 3 above)
Comment (AmbientTea) [reply CTL_Scripts_Reference.md]: Me can use mamba-cli aka. sc-evm-cli.

File: CTL_Scripts_Reference.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+}
+```
+
+### Register committee candidate
+
+In order to generate the signatures, you can use the `sc-evm-cli` utility:
+
+```shell
+sc-evm-cli generate-signature \
+      --genesis-hash "5f4a33beaf8876b9697fb567c9af11390434086a10454c32091e2017366dcd7a" \
```
Comment (LGLO) [inline CTL_Scripts_Reference.md:127]: You don't have to double quote params, but it's not harmful to do so

File: CTL_Scripts_Reference.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (LGLO) [reply CTL_Scripts_Reference.md]:
```
ctl-main claim --payment-signing-key-file ../takao2/lglo.skey --combined-proof d8799fd8799f0509581d606e9d4f6a3f900f7b510b27f29d8e79c550686ce093946b23b3d1828ed8799f5820695b174994f178191287bd8f4189ac7d6b92897244adab4e5ffb5bc078252cc0ffff9fd8799f0058202fba7440e3a151f212d2602bbbbbd3819ff8e6e775b5317d5dd4813034b161b6ffd8799f015820c72ebdc0ecc293ad4f5b28618f54e00d38868ee8f6c310bed6f7f4b4de7ec31affd8799f0058207198c011dbeb151bd5ed7afc2818549a5189f462eaa0b8d4499c77daac3bd030ffffff
[INFO] 2023-01-17T09:53:30.317Z FUELMintingPolicy.runFuelMP: Submitted Tx: (TransactionHash (hexToByteArrayUnsafe "1a724b659b7fd48366fd0fb3b883d38589a9624215f8b37e5a21b75a27d3f49c"))
[INFO] 2023-01-17T09:54:04.569Z FUELMintingPolicy.runFuelMP: Tx submitted successfully!
{"endpoint":"ClaimAct","transactionId":"1a724b659b7fd48366fd0fb3b883d38589a9624215f8b37e5a21b75a27d3f49c"}
```
my example

File: CTL_Scripts_Reference.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+  --registration-utxo 7eddcb7807899d5078ebc25c59d372b484add88604db461e6ef077fd0379733d#0
+```
+
+### Deregister committee candidate
+
+```shell
+node main.js -- deregister --spo-public-key aabbcc
+```
+
+### Committee hash update
```
Comment (LGLO) [inline CTL_Scripts_Reference.md]: Could you drop a line that it's required to add a `--committee-pub-key-and-signature` for each committee member, even if it hasn't signed. I know it's implicitly shown in this example.

File: Trustless_Sidechain_CTL_Reference.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+```bash
+export SIGNING_KEY=<path to file>/server.skey
+```
+
+Available commands:
+
+```
+  init                     Initialise sidechain
+  addresses                Get the script addresses for a given sidechain
+  mint*                     Mint a certain amount of SC_Token tokens
```
Comment (vvviiimmm) [inline Trustless_Sidechain_CTL_Reference.md:44]: should we even mention it?

File: Trustless_Sidechain_CTL_Reference.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+To get the script addresses for a given sidechain, you can use the following command:
+
+```shell
+node main.js -- addresses 
+```
+
+### Burn user owned SC_Token tokens
+
+```shell
+node main.js -- burn --amount 5 --recipient aabbcc
```
Comment (vvviiimmm) [inline Trustless_Sidechain_CTL_Reference.md:104]: valid address here?

File: Trustless_Sidechain_CTL_Reference.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+In order to generate the signatures, you can use the `sc-evm-cli` utility:
+
+```shell
+sc-evm-cli generate-signature \
+      --genesis-hash "5f4a33beaf8876b9697fb567c9af11390434086a10454c32091e2017366dcd7a" \
+      --sidechain-id 123 \
+      --registration-utxo  "9713f220d6d388a63fe3f426212da3af063372215e8e764c470225eab0384556#1" \
+      --spo-signing-key "spo_prv_key" \
+      --sidechain-signing-key "sc_prv_key" \
```
Comment (vvviiimmm) [inline Trustless_Sidechain_CTL_Reference.md:131]: valid values here can be helpful

**Review by vvviiimmm — APPROVED** [review]
just nits. As a user I'd like to see some real values that a tool can 'parse' (sometimes you can pass garbage and move forward, instead of figuring out the expected format)

File: CTL_Scripts_Reference.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
+# Using CTL Scripts
+
+## Hardware and OS Requirements
+CTL Scripts should run on any modern OS and hardware combination capable of running a Node installation:
+- Linux
+- MacOS
+- Windows 10+
```
[the remark this answers is not in the document]
Comment (AmbientTea) [reply CTL_Scripts_Reference.md]: WSL man, Windows is a real OS now.

File: CTL_Scripts_Reference.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+  --sidechain-public-key 03eef26d3cf978e0fc2d786c443b1284b27b265a7c82eeeec68c24cd3fd0bb6428 \
+  --spo-signature 980db1db31457189326e948c7f292b16278ab91bd45f5fd6ee9ad637bf993f26936c17ee126e510c52d0a3381b52acb36a2a89d4fe55a587cf3478678114dd0f \
+  --sidechain-signature bd00090a8a26b3aad534ae2e75ce4ab5b284ffad0751b793d447a3980f770f217929543c21bc7d2567c6ee0c23b983e3983f22d8eb41dfb901a6c31ae3d5b41d \
+  --registration-utxo 7eddcb7807899d5078ebc25c59d372b484add88604db461e6ef077fd0379733d#0
+```
+
+### Deregister committee candidate
+
+```shell
+node main.js -- deregister --spo-public-key aabbcc
```
[the remark this answers is not in the document]
Comment (AmbientTea) [reply CTL_Scripts_Reference.md]: No, I missed it.

**Pull request merged**